packages/gatsby/src/schema/types/date.js

```diff
@@ -99,8 +99,13 @@ const GraphQLDate = new GraphQLScalarType({
 // Check if this is a date.
 // All the allowed ISO 8601 date-time formats used.
 function isDate(value) {
+  // quick check if value does not look like a date
+  if (typeof value === `number` || !/^\d{4}/.test(value)) {
```

Comment:
I'm not sure an inline regex is the fastest way of determining if the first 4 values are numeric.

You could pre-compile, or alternatively even have a set of possible single digit strings, and iterate over the first 4 and break if it's not in the set.

Reply:
According to this benchmark, it seems inline regex or pre-compiled is actually fastest. 

https://jsperf.com/check-whether-string-starts-with-four-numbers

In Chrome it doesn't appear to matter if it's inline or not, in Safari I notice a 2% improvement in pre-compiling.

Reply:
I tried both ways and the change was marginal, about 600s to build the schemas. Without the patch, it was 8000s. My guess is that the regex is optimized automatically.

Reply:
@me4502 I used string.slice(0, 4) and checked if it was a number and regex was faster. I'm pretty sure v8 optimizes for it.

@millette at least we got somewhere, not fast enough but faster.